### PR TITLE
Updated Nuget packages for all projects

### PR DESCRIPTION
**Pull request by GintasS**
## Proposed changes
- Updated NuGet packages for all projects.
- Fixed failing AppVeyor integration tests.

## Test methodology 

- Fixed failing tests.
- Used existing tests to ensure quality.

## Test environment(s) 

GIT 2.21.0.windows.1
Windows 10

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: Packages.props [hunk 1]
```diff
@@ -2,32 +2,32 @@
...
     <PackageReference Update="Microsoft-WindowsAPICodePack-Shell" Version="1.1.4" />
-    <PackageReference Update="Newtonsoft.Json" Version="12.0.3" />
-    <PackageReference Update="NString" Version="1.2.0" />
-    <PackageReference Update="RestSharp" Version="106.11.4" />
-    <PackageReference Update="SmartFormat.NET" Version="2.5.0" />
+    <PackageReference Update="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Update="NString" Version="2.1.1" />
+    <PackageReference Update="RestSharp" Version="106.12.0" />
+    <PackageReference Update="SmartFormat.NET" Version="2.7.0" />
     <PackageReference Update="StyleCop.Analyzers" Version="1.2.0-beta.321" />
```
Comment (mstv) [inline Packages.props:24]: Are the `StyleCop.Analyzers` really obsolete? Do we need to migrate the regarding settings?

**Review by mstv — COMMENTED** [review]
Please mention removed packages in the PR description.

File: Packages.props [hunk 2]
```diff
@@ -2,32 +2,32 @@
...
   <ItemGroup>
     <!-- Solution dependencies -->
-    <PackageReference Update="AdysTech.CredentialManager" Version="2.1.1" />
-    <PackageReference Update="Atlassian.SDK" Version="12.1.1" />
-    <PackageReference Update="AppInsights.WindowsDesktop" Version="2.13.1" />
-    <PackageReference Update="Ben.Demystifier" Version="0.1.6" />
-    <PackageReference Update="ConEmu.Core" Version="21.03.14" />
-    <PackageReference Update="EnvDTE" Version="16.9.31023.347" />
+    <PackageReference Update="AdysTech.CredentialManager" Version="2.3.0" />
+    <PackageReference Update="Atlassian.SDK" Version="12.4.0" />
```
Comment (gerhardol) [inline Packages.props:6]: Note: Will have to be tested by plugins

**Review by gerhardol — COMMENTED** [review]
Quoting mstv: LGTM, have not run

> GIT 2.21.0.windows.1

Please update Git...

**Review by RussKie — APPROVED** [review]

**Pull request merged**